pydicom/filebase.py

```diff
@@ -30,19 +30,13 @@ def __init__(self, *args, **kwargs):
     def read_le_tag(self):
         """Read and return two unsigned shorts (little endian)
            from the file."""
-        bytes_read = self.read(4)
-        if len(bytes_read) < 4:
-            # needed for reading "next" tag when at end of file
-            raise EOFError
+        bytes_read = self.read(4, need_exact_length=True)
         return unpack(b"<HH", bytes_read)
 
     def read_be_tag(self):
```

Comment:
a) agreed that this would be better, but I don't want to break the API; the comment I will fix of course
b) we have no test data for compressed big endian, I'm not sure where to find it given that Big Endian is retired - maybe we have to generate it, but I would leave that for later